opensrp-giz-malawi/src/main/java/org/smartregister/giz/recievers/SyncStatusReciever.java

```diff
@@ -0,0 +1,40 @@
+package org.smartregister.giz.recievers;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.giz.job.GizReProcessJob;
+
+import java.io.Serializable;
+
+import timber.log.Timber;
+
+import static org.smartregister.receiver.SyncStatusBroadcastReceiver.EXTRA_FETCH_STATUS;
+
+public class SyncStatusReciever  extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Bundle data = intent.getExtras();
+        if (data != null) {
+            Serializable fetchStatusSerializable = data.getSerializable(EXTRA_FETCH_STATUS);
+            if (fetchStatusSerializable instanceof FetchStatus) {
+                FetchStatus fetchStatus = (FetchStatus) fetchStatusSerializable;
+                    if (fetchStatus.equals(FetchStatus.fetched)) {
+                        GizReProcessJob.scheduleJobImmediately(GizReProcessJob.TAG);
```

Comment:
I believe we want to call the job once sync is complete eg sync is done fetching all the latest records and not successfully fetched one batch. I'm not sure if this is what is currently implemented